runtime/decimal.c

```diff
@@ -129,6 +131,21 @@ TaggedPtr _bal_decimal_neg(TaggedPtr tp) {
     return createDecimal(&d);
 }
 
+double _bal_decimal_to_float(TaggedPtr tp) {
+    char dblStr[DECQUAD_String];
+    const decQuad *d = taggedToDecQuad(tp);
+    decQuadToString(d, dblStr);
+    double dbl;
+    // It is garanteed that dblStr has the correct format,
+    // because decQuadToString returns scientific notation.
+    // Therefore it is not needed to handle the EOF.
+    sscanf(dblStr, "%lf", &dbl);
+    if (isfinite(dbl)) {
+        return dbl;
+    }
+    return decQuadClass(d) == DEC_CLASS_POS_NORMAL ? DBL_MAX : -DBL_MAX;
```

Comment:
Instead of `isinf`, decQuadClass is used here, because `isinf` is cannot identify -inf over +inf in macos.

Reply:
Why do you need to do something special if `isfinite` is false?

Reply:
It is not needed, Removed.